NDTensors/test/blocksparse.jl

```diff
@@ -9,7 +9,12 @@ using GPUArraysCore: @allowscalar
   devs = devices_list(copy(ARGS))
 
   @testset "test device: $dev" for dev in devs
-    elt = (dev == NDTensors.mtl ? Float32 : Float64)
+    if dev == NDTensors.mtl
+      elt = Float32
+      NDTensors.default_eltype() = Float32
```

Comment:
This is not a good idea.

Reply:
A pattern I used for this can be seen here:
https://github.com/ITensor/ITensors.jl/blob/408516dce2e0182e04a6fb27b501b04a31b630fc/NDTensors/test/ITensors/TestITensorDMRG/runtests.jl

Reply:
So I was going to do something like this but one problem I was running into is that constructors which do not specify the eltype like `Tensor(undef, (2,4))`, make a Float64 tensor and then when I call `NDTensors.mtl` the function to convert a Float64 tensor which fails. My thought was that, since we have our constructors initialize Tensors' eltype to `NDTensors.default_eltype()` we could modify the default eltype for the time being. 
To keep the future in mind, I think this problem will be mitigated when we convert the `default_eltype` to `UnspecifiedType`

Reply:
It seems like a better solution is to call more general constructors `Tensor{elt}(undef, (2, 4))`, `randomTensor(elt, (2, 4))`, etc. , and if those don't exist then they shouldn't be hard to define. I think most constructors support passing the element type in some way at this point.

Otherwise we can use `adapt(elt, a)` after the tensor is created already.

I really don't think we should be controlling this in a global way, even in tests. That could cause some very confusing behavior for people, for example if they run the tests and then they run some other code and suddenly their element types are all changed.

Reply:
I had written this up previously using adapt but I was worried about the same problem (i.e. changing types without users knowing whats going on or why we are using adapt) so I think using `Tensor{elt}` makes the most sense and is the most transparent for users. I can start making the changes. Thanks for the input!